alpakka/src/main/scala/org/scanamo/ops/retrypolicy/RetryPolicy.scala

```diff
@@ -0,0 +1,76 @@
+package org.scanamo.ops.retrypolicy
+
+import java.util.concurrent.ScheduledExecutorService
+
+import org.scanamo.ops.retrypolicy.RetryPolicy._
+
+import scala.concurrent.duration.FiniteDuration
+
+sealed abstract class RetryPolicy extends Product with Serializable { self =>
+  final def maybeScheduler: Option[ScheduledExecutorService] = self match {
+    case policy @ Constant(_, _)       => Option(policy.scheduler)
+    case policy @ Linear(_, _, _)      => Option(policy.scheduler)
+    case policy @ Exponential(_, _, _) => Option(policy.scheduler)
+    case And(thisPolicy, thatPolicy)   => thisPolicy.maybeScheduler orElse thatPolicy.maybeScheduler
+    case Or(thisPolicy, thatPolicy)    => thisPolicy.maybeScheduler orElse thatPolicy.maybeScheduler
+    case _                             => None
+  }
+
+  final def done: Boolean = self match {
```

Comment:
I had a flash of insight yesterday. This `done` is false. If you have `And(Max(1), Constant(2.s))` it will keep on going infinitely, even after the max becomes `Max(0)`. What we need instead is
```scala
def continue = self match {
  case Max(0)    => false
  case And(x, y) => x.continue && y.continue
  case Or(x, y)  => x.continue || y.continue
  case _         => true
}
```

Remember to remove `numberOfRetries` from `Linear`, `Constant` and `Exponential`.

Reply:
Oh, wait, actually no it works with `done` since the test is negated! Sorry about that. Let's keep it as it is then, with only `numberOfRetries` to go away.

Reply:
@regiskuckaertz 

> I had a flash of insight yesterday. This done is false. If you have And(Max(1), Constant(2.s)) it will keep on going infinitely, even after the max becomes Max(0).

I guess it'll not go on infinitely based on the initial dry run without writing test cases. `And(Max(1), Constant(2.s))` will finally get to `And(Max(0), Constant(2.s))` because of `And(thisPolicy.update, thatPolicy.update)` which will update the Max(1) to Max(0) and then 
`case And(thisPolicy, thatPolicy)        => thisPolicy.done && thatPolicy.done`
will give true and true respectively which will end the loop. I hope I did not miss anything :)

> Remember to remove numberOfRetries from Linear, Constant and Exponential.
If retries with delay as 0 is not a requirement, which is what happens in case of using Max, then Max doesn't look like a good choice and numberOfRetries can stay in Linear, Constant and Exponential. What do you think?

Reply:
I'd like to keep it as the documentation states that for internal server errors, it is safe to retry without waiting.